src/output.zig

```diff
@@ -128,27 +131,26 @@ pub const Source = struct {
         if (!stdout_stream_set) {
             stdout_stream_set = true;
             if (comptime Environment.isNative) {
-                var is_color_terminal: ?bool = null;
-                if (_source.stream.isTty()) {
+                var enable_color: ?bool = null;
+                if (isForceColor()) {
+                    enable_color = true;
+                } else if (isNoColor() or !isColorTerminal()) {
+                    enable_color = false;
+                }
+
+                const is_stdout_tty = _source.stream.isTty();
+                if (is_stdout_tty) {
                     stdout_descriptor_type = OutputStreamDescriptor.terminal;
-                    enable_ansi_colors_stdout = isColorTerminal();
-                    is_color_terminal = enable_ansi_colors_stdout;
-                } else if (isForceColor()) |val| {
-                    enable_ansi_colors_stdout = val;
-                } else {
-                    enable_ansi_colors_stdout = false;
                 }
 
-                if (_source.error_stream.isTty()) {
+                const is_stderr_tty = _source.error_stream.isTty();
+                if (is_stderr_tty) {
                     stderr_descriptor_type = OutputStreamDescriptor.terminal;
-                    enable_ansi_colors_stderr = is_color_terminal orelse isColorTerminal();
-                } else if (isForceColor()) |val| {
-                    enable_ansi_colors_stderr = val;
-                } else {
-                    enable_ansi_colors_stderr = false;
                 }
 
-                enable_ansi_colors = enable_ansi_colors_stderr or enable_ansi_colors_stdout;
+                enable_ansi_colors_stdout = enable_color orelse is_stdout_tty;
+                enable_ansi_colors_stderr = enable_color orelse is_stderr_tty;
```

Comment:
I don't think this is correct

TTYs do not all support colors

Reply:
If the tty does not support color, `enable_color` is set to `false`.

Reply:
TTYs declare their support for color, the default is false

Reply:
if `enable_color` is `null`, then it will be enabled if `is_stderr_tty` is defined

`orelse` in zig is `or` but only for `null`, like `??` in JS

Reply:
> if enable_color is null, then it will be enabled if is_stderr_tty is true

That's intended

```zig
var enable_color: ?bool = null;
if (isForceColor()) {
  enable_color = true;
} else if (isNoColor() or !isColorTerminal()) {
  enable_color = false;
}
const is_stdout_tty = _source.stream.isTty();
```

When a terminal does not support colors...
*  `enable_color` starts off as `null`
* `isForceColor()` evaluated to `false`, continue
* `isNoColor()` evaluated to `false`, continue
* `isColorTerminal()` evaluated to `false`, but the statement is negated, so it evaluated to `true`, step
  * `enable_color = false;` is evaluated

In this case, `enable_color` is a tri-state, it is only `null` when there is no `NO_COLOR` or `FORCE_COLOR` variable _and_ the terminal accepts color. Let me know if that makes sense.

Would `force_color` be a better variable name? I get the sense maybe it's confusing.